test/data-provider.test.js

```diff
@@ -563,6 +564,33 @@ describe('data provider', () => {
   });
 });
 
+describe('attached', () => {
+  it('should have rows when attached and shown after cache is cleared on hidden grid', (done) => {
+    const grid = window.document.createElement('vaadin-grid');
+    const col = window.document.createElement('vaadin-grid-column');
+    col.setAttribute('path', 'item');
+    grid.appendChild(col);
+
+    grid.size = 1;
+    grid.dataProvider = function (params, callback) {
+      callback([{ item: 'A' }]);
+    };
+
+    grid.style.display = 'none';
+    window.document.body.appendChild(grid);
+
+    setTimeout(() => {
```

Comment:
nit: please consider using `await aTimeout(0)` and make test  `async () =>` instead of `(done) =>`

Reply:
Updated.